### PR TITLE
support atf in install space

### LINKED ISSUE #175: atf cannot be used from install space
because the cmake macro tries to run `generate_tests.py` from `@(CMAKE_SOURCE_DIR)/scripts/generate_tests.py`

```
  Errors     << mojin_atf_kevin:cmake /root/target_ws/logs/mojin_atf_kevin/build.cmake.000.log
  ATF: executing test generation macro with 'atf/test_generation_config.yaml' and EXECUTE_TESTS=True
  /usr/bin/python2: can't open file '/root/git/app_ws/src/atf/atf_core/scripts/generate_tests.py': [Errno 2] No such file or directory
  CMake Error at /root/git/app_ws/install/share/atf_core/cmake/atf_core-extras.cmake:39 (message):
    -- ATF: generating test files failed: exit_code='2'
  Call Stack (most recent call first):
    CMakeLists.txt:27 (atf_test)
```

### PR DESCRIPTION
**Pull request by floweisshardt**
fixes https://github.com/floweisshardt/atf/issues/175

### CODE REVIEW

File: atf_core/cmake/atf_core-extras.cmake.em [hunk 1]
```diff
@@ -1,4 +1,12 @@
-set(generate_tests_script @(CMAKE_SOURCE_DIR)/scripts/generate_tests.py)
+if(EXISTS "@(CMAKE_INSTALL_PREFIX)/lib/atf_core/generate_tests.py")
```
Comment (fmessmer) [inline atf_core/cmake/atf_core-extras.cmake.em]:
```suggestion
if(EXISTS "@(CMAKE_INSTALL_PREFIX)/lib/atf_core/generate_tests.py")   # atf_core uses install space
```

File: atf_core/cmake/atf_core-extras.cmake.em [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
+    set(generate_tests_script @(CMAKE_INSTALL_PREFIX)/lib/atf_core/generate_tests.py)
+    message("ATF: install space found. generate_tests_script=${generate_tests_script}")
+elseif(EXISTS "@(CMAKE_SOURCE_DIR)/scripts/generate_tests.py")                    # 2. atf_core and atf_test in same workspace (devel)
```
Comment (fmessmer) [inline atf_core/cmake/atf_core-extras.cmake.em]:
```suggestion
elseif(EXISTS "@(CMAKE_SOURCE_DIR)/scripts/generate_tests.py")   # atf_core uses devel space
```

File: atf_core/cmake/atf_core-extras.cmake.em [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+    set(generate_tests_script @(CMAKE_SOURCE_DIR)/scripts/generate_tests.py)
+    message("ATF: devel space found. generate_tests_script=${generate_tests_script}")
+else()
```
Comment (fmessmer) [inline atf_core/cmake/atf_core-extras.cmake.em]:
```suggestion
else()   # atf_core not found
```

File: atf_core/cmake/atf_core-extras.cmake.em [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
+    message(FATAL_ERROR "ATF: no install and no devel space found. CMAKE_SOURCE_DIR=@(CMAKE_SOURCE_DIR), CMAKE_INSTALL_PREFIX=@(CMAKE_INSTALL_PREFIX)")
```
Comment (fmessmer) [inline atf_core/cmake/atf_core-extras.cmake.em]:
```suggestion
    message(FATAL_ERROR "ATF: atf_core not found in install or devel space. CMAKE_SOURCE_DIR=@(CMAKE_SOURCE_DIR), CMAKE_INSTALL_PREFIX=@(CMAKE_INSTALL_PREFIX)")
```

**Review by fmessmer — CHANGES_REQUESTED** [review]
if you add comments, have a look at my suggestions

I'll prepare a PR for `mojin_atf` to test this feature branch....I'd suggest to hold merging until we confirm it wors for `mojin_atf`

**Comment by fmessmer** [comment]
ah, we can only test this properly when this atf feature branch is used in our docker images
I'll prepare something

**Comment by benmaidel** [comment]
Strange that there is nowhere any documentation about the @ syntax. Is this perhaps something special from the `catkin_package(CFG_EXTRAS...` function?

**Pull request merged**